library/bootloader_facts.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2023, Sergei Petrosian <spetrosi@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: bootloader_facts
+
+short_description: Gather information for kernels as Ansible facts
+
+version_added: "0.0.1"
+
+description:
+    - "WARNING: Do not use this module directly! It is only for role internal use."
+    - Gather information for kernels as Ansible facts
+
+author:
+    - Sergei Petrosian (@spetrosi)
+"""
+
+EXAMPLES = r"""
+# Run with no parameters to gather facts
+bootloader_facts:
+"""
+
+import re
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict()
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # changed is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(changed=False, ansible_facts=dict(bootloader_facts=dict()))
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    # Keys from `grubby --info=ALL`, values for which this module returns
+    grubby_keys = ["args", "id", "index", "initrd", "kernel", "root", "title"]
+
+    rc, stdout, stderr = module.run_command("grubby --info=ALL")
+    if "Permission denied" in stderr:
+        module.fail_json(msg="You must run this as sudo", **result)
+    kernel_count = len(re.findall("index=\d+",stdout))
+    stdout_lines = stdout.strip().split('\n')
+    kernels = []
+    index_count = 0
+    while len(kernels) < kernel_count:
+        for line in stdout_lines:
+            if re.search("index=\d+", line):
+                index_count += 1
+            if len(kernels) < index_count:
```

Comment:
```suggestion
```

Reply:
This is needed because we need to append an empty `dict` to kernels only when loop meets a new `index=\d`. Otherwise, `kernels` has the number of dicts of `len(stdout_lines)`

Reply:
I don't understand - if you only add a new `dict` to the `kernels` list when `re.search("index=\d+", line)`, then that should be right.

Reply:
Riiight, now I see, that's true. Thanks, I'll apply this.